### PR TITLE
feat(unclaim): unclaim an object by an owner

### PR DESCRIPTION
**Pull request by daddykotex**
I also added some it tests that will be run when merged against mnubo/smartobjects-python-client
 by travis.

### CODE REVIEW

File: ittests/it_test.py [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+        }
+
+    @staticmethod
+    def eventuallyAssert(myAssert):
+        TestHelper.eventuallyAssertWithDelay(myAssert, TestHelper.timeout, TestHelper.delay)
+
+    @staticmethod
+    def eventuallyAssertWithDelay(myAssert, timeout, delay):
+        if (timeout < delay):
+            raise Exception("timeout should be bigger than delay")
```
Comment (mvonthron) [inline ittests/it_test.py]: `ValueError`

File: ittests/it_test.py [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+
+        current_time = lambda: int(round(time.time()))
```
Comment (mvonthron) [inline ittests/it_test.py]: unify the casing between `snake_case` and `mixedCase` (usually `snake_case` in Python)

File: ittests/it_test.py [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+                myAssert()
+                return
+            except AssertionError as e:
+                lastAssertException = e
+                print "Assert failed: {}. Will retry".format(e)
+            except Exception as e:
+                raise e
+            time.sleep(delay)
+
+        if lastAssertException != None:
```
Comment (mvonthron) [inline ittests/it_test.py]: `None` should be compared with `is not`:

```python
if lastAssertException is not None:
    pass
```

File: ittests/it_test.py [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+        stopper = current_time() + timeout
+        lastAssertException = None
+
+        while current_time() < stopper:
+            try:
+                myAssert()
+                return
+            except AssertionError as e:
+                lastAssertException = e
+                print "Assert failed: {}. Will retry".format(e)
```
Comment (mvonthron) [inline ittests/it_test.py]: You should use Python `logging` to avoid polluting the output.

File: ittests/it_test.py [hunk 5]
```diff
@@ -0,0 +1,123 @@
...
+
+        current_time = lambda: int(round(time.time()))
+
+        stopper = current_time() + timeout
+        lastAssertException = None
+
+        while current_time() < stopper:
+            try:
+                myAssert()
+                return
```
Comment (mvonthron) [inline ittests/it_test.py]: Maybe there's a way to use a `lastAssertException` or a similar flag to have a cleaner exit.

File: ittests/tests_owners.py [hunk 6]
```diff
@@ -0,0 +1,117 @@
...
+
+      with self.assertRaises(ValueError):
+        self.client.owners.delete(usernameToDelete)
+
+      self.client.owners.create({
+          "username": usernameToDelete,
+          "x_password": "password-{}".format(currentUUID),
+      })
+
+      def searchOwnerCreated():
```
Comment (mvonthron) [inline ittests/tests_owners.py]: The Python style is usually to use `snake_case`:
  https://www.python.org/dev/peps/pep-0008/#function-names

Comment (mvonthron) [inline ittests/tests_owners.py]: (I know... `assertEqual`...)

But at least we should be consistent within a package.

File: README.md [hunk 7]
```diff
@@ -279,7 +281,7 @@ For complex queries, it is recommended to use this feature to reduce errors.
 
 ```python
 validation_result = client.search.validate_query({
-    "fro": "owner",
```
Comment (daddykotex) [inline README.md]: oh wow did not see that

File: ittests/it_test.py (hunk 5 above)
Comment (daddykotex) [inline ittests/it_test.py]: what do you mean? when you reach this line, it means the assert was successful, the `lastAssertException` is irrelevant at this point. It contains the failure of the last attempt

File: build.sh [hunk 8]
```diff
@@ -0,0 +1,9 @@
+#!/bin/bash
+
+if [ "$TRAVIS_PULL_REQUEST" = "false" ]; then
```
Comment (mvonthron) [inline build.sh]: No they're hidden in the log (but you can _choose_ to show them). 

>Define variables in the Repository Settings that:
> ...
> contain sensitive data, such as third-party credentials.

File: ittests/it_test.py (hunk 4 above)
Comment (daddykotex) [inline ittests/it_test.py]: fixing

File: ittests/it_test.py (hunk 3 above)
Comment (daddykotex) [inline ittests/it_test.py]: fixing

File: ittests/tests_owners.py (hunk 6 above)
Comment (daddykotex) [inline ittests/tests_owners.py]: ill make everything snake_case

File: ittests/it_test.py (hunk 1 above)
Comment (daddykotex) [inline ittests/it_test.py]: fixing

File: ittests/it_test.py (hunk 2 above)
Comment (daddykotex) [inline ittests/it_test.py]: fixing

**Pull request merged**